sources/packages/api/src/route-controllers/assessment/assessment.system-access.controller.ts

```diff
@@ -0,0 +1,309 @@
+import {
+  Controller,
+  Get,
+  NotFoundException,
+  Param,
+  HttpStatus,
+  UnprocessableEntityException,
+  Patch,
+  Body,
+  Post,
+} from "@nestjs/common";
+import {
+  ASSESSMENT_INVALID_OPERATION_IN_THE_CURRENT_STATE,
+  ASSESSMENT_NOT_FOUND,
+  DisbursementScheduleService,
+  EducationProgramOfferingService,
+  StudentAssessmentService,
+} from "../../services";
+import BaseController from "../BaseController";
+import { AuthorizedParties } from "../../auth/authorized-parties.enum";
+import { ApiResponse, ApiTags } from "@nestjs/swagger";
+import {
+  ApplicationAssessmentDTO,
+  CreateDisbursementsDTO,
+  UpdateAssessmentDataDTO,
+  UpdateAssessmentStatusDTO,
+  UpdateAssessmentWorkflowIdDTO,
+  UpdateProgramInfoDTO,
+} from "./models/assessment.system-access.dto";
+import { AllowAuthorizedParty } from "../../auth/decorators";
+import { ClientTypeBaseRoute } from "../../types";
+
+@AllowAuthorizedParty(AuthorizedParties.formsFlowBPM)
+@Controller("assessment")
+@ApiTags(`${ClientTypeBaseRoute.SystemAccess}-assessment`)
+export class AssessmentSystemAccessController extends BaseController {
+  constructor(
+    private readonly assessmentService: StudentAssessmentService,
+    private readonly offeringService: EducationProgramOfferingService,
+    private readonly disbursementScheduleService: DisbursementScheduleService,
+  ) {
+    super();
+  }
+
+  /**
+   * Get the assessment and the related application information.
+   * This method is used by the the assessment workflow as a main source
+   * of information for the assessment/reassessment and the application.
+   * @param assessmentId assessment id .
+   * @returns assessment and the related application information.
+   */
+  @Get(":id")
+  @ApiResponse({
```

Comment:
Are we planning to have this in our newly created methods? or as discussed can remove and do it in a common place?

Reply:
During the discussion, my understanding was that whatever is part of the body of the controller is suitable to be added as API response and also the OK/CREATED when we would provide some description.